src/main/java/edu/ie3/datamodel/io/source/IdCoordinateSource.java

```diff
@@ -5,48 +5,86 @@
 */
 package edu.ie3.datamodel.io.source;
 
+import edu.ie3.datamodel.io.factory.SimpleFactoryData;
+import edu.ie3.datamodel.io.factory.timeseries.IdCoordinateFactory;
 import edu.ie3.util.geo.CoordinateDistance;
 import edu.ie3.util.geo.GeoUtils;
 import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.apache.commons.lang3.tuple.Pair;
 import org.locationtech.jts.geom.Point;
 
 /**
  * This class serves mapping purposes between the ID of a coordinate and the actual coordinate with
  * latitude and longitude values, which is especially needed for data source that don't offer
  * combined primary or foreign keys.
  */
-public interface IdCoordinateSource extends DataSource {
+public class IdCoordinateSource implements DataSource {
```

Comment:
For weather coordinates, we have some relevant changes in #694, something we should keep in mind here. I think the changes can be combined though.